### ISSUE TITLE
Restrict SFTP access to user's $HOME directory (or use blacklist)

### ISSUE THREAD

**Issue by fevangelou** [netblue30/firejail#2400]
First off, thank you for such a great piece of software.

On a multi-tenant server setup using Ubuntu 18.04, I have managed to restrict SSH access for users so they are not allowed to see certain folder (e.g. /backup, /etc/nginx, /etc/php and so on) by swapping their shell with "/usr/bin/firejail" upon user creation. Aside the aforementioned folders (which are listed in my custom /etc/firejail/disable-common.local), these users cannot navigate to other /home/* folders and processes like top/htop are nicely restricted to these users' processes. So all good on the SSH front.

However, for practical reasons, users must have SFTP access as well. I've found no clue so far as to how/if it's possible to restrict SFTP access similarly to SSH access.

In other words, would it be possible to always enforce /usr/lib/sftp-server or /usr/lib/openssh/sftp-server (whichever is mentioned in /etc/ssh/sshd_config at "Subsystem sftp...") to run via Firejail?

Thanks in advance.

**Comment by chiraag-nataraj** [comment]
Why not just set the option to `firejail /usr/lib/sftp-server` or whatever in `/etc/ssh/sshd_config`? The other alternative (if you're spawning the server as a system service) would be to edit the init script or systemd unit file to launch the program via `firejail`.

**Comment by Gimly86** [comment]
Hello all,

Same question here, I test by launching this command : 
`/usr/bin/firejail --profile=/etc/firejail/ssh.proile -c /usr/lib/openssh/sftp-server -l INFO`

I have an can't connect to my SFTP client with this error:

`subsystem request for sftp by user testsftp failed, subsystem not found`

Same problem with --noprofile and same problem with this in /etc/ssh/sshd_config : 

`Subsystem /usr/bin/firejail --profile=/etc/firejail/ssh.profile /usr/lib/openssh/sftp-server -l INFO`

What is the best practices to firejail openssh SFTP server, please ?

Thanks in advance for your help !

**Comment by fevangelou** [comment]
@chiraag-nataraj Did so, but SFTP completely craps out. Even moved the command to a shell script in case the syntax was causing issues. SFTP breaks entirely.

There must be a better way to do this. I'm surprised SFTP is not covered in the docs, blog posts etc.

**Comment by rusty-snake** [comment]
IMHO the best is:
```
###Subsystem sftp /usr/lib/openssh/sftp-server
# Enable to built-in implementation of SFTP
Subsystem sftp internal-sftp

# This section must be placed at the very end of sshd_config
Match Group exchangefiles
  # Force the connection to use the built-in SFTP support
  ForceCommand internal-sftp
  # Chroot the connection into the home directory of the user being authenticated
  ChrootDirectory %h
…
```
https://wiki.ubuntuusers.de/SSH/

**Comment by rusty-snake** [comment]
I'm closing here due to inactivity, please fell free to reopen if you have more questions.
